client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx

```diff
@@ -136,139 +119,166 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
+                  target="_blank"
+                  rel="noopener noreferrer"
+                >
+                  {jurisdictionFile && jurisdictionFile.file
+                    ? jurisdictionFile.file.name
+                    : ''}
+                </a>
+              </td>
             </tr>
-          ))}
-        </tbody>
-      </ContestsTable>
-      <ContestsTable>
-        <thead>
-          <tr>
-            <th>Opportunistic Contests</th>
-            <th>Jurisdictions</th>
-          </tr>
-        </thead>
-        <tbody>
-          {opportunisticContests.map((c: IContest) => (
-            <tr key={c.id}>
-              <td>{c.name}</td>
-              <td>{c.jurisdictionIds.join(', ')}</td>
+            <tr>
+              <td>Audit Board Data Entry:</td>
+              <td>{online ? 'Online' : 'Offline'}</td>
             </tr>
-          ))}
-        </tbody>
-      </ContestsTable>
-      <H4>Sample Size Options</H4>
+          </tbody>
+        </SettingsTable>
+        <ContestsTable>
+          <thead>
+            <tr>
+              <th>Target Contests</th>
+              <th>Jurisdictions</th>
+            </tr>
+          </thead>
+          <tbody>
+            {targetedContests.map((c: IContest) => (
+              <tr key={c.id}>
+                <td>{c.name}</td>
+                <td>{c.jurisdictionIds.join(', ')}</td>
+              </tr>
+            ))}
+          </tbody>
+        </ContestsTable>
+        <ContestsTable>
+          <thead>
+            <tr>
+              <th>Opportunistic Contests</th>
+              <th>Jurisdictions</th>
+            </tr>
+          </thead>
+          <tbody>
+            {opportunisticContests.map((c: IContest) => (
+              <tr key={c.id}>
+                <td>{c.name}</td>
+                <td>{c.jurisdictionIds.join(', ')}</td>
+              </tr>
+            ))}
+          </tbody>
+        </ContestsTable>
+      </ElevatedCard>
+      <br />
+      <H4>Estimated Sample Sizes</H4>
       <Formik
         initialValues={{
-          sampleSize: sampleSizeOptions.length ? sampleSizeOptions[0].size : '',
-          customSampleSize: '',
+          sampleSizes: initialValues,
+          customSampleSizes: initialCustomValues,
         }}
         enableReinitialize
-        onSubmit={v => {
-          if (v.sampleSize === 'custom') {
-            setSampleSize(v.customSampleSize)
-          } else {
-            setSampleSize(v.sampleSize)
-          }
+        onSubmit={({ sampleSizes: sizes, customSampleSizes }) => {
+          setSampleSizes(
+            Object.keys(sizes).reduce((a, contestId) => {
+              if (sizes[contestId] === 'custom') {
+                return {
+                  ...a,
+                  [contestId]: customSampleSizes[contestId],
+                }
+              }
+              return { ...a, [contestId]: sizes[contestId] }
+            }, {})
+          )
```

Comment:
i notice that you use this pattern a few times. I've seen it abstracted out into a helper function called `mapValues` in previous codebases I've worked on, since it comes up a lot. that would look like this:

```ts
const mapValues = (obj, fn) =>
  Object.keys(obj).reduce((a, key) => ({
    ...a,
    [key]: fn(obj[key], key)
  }), {})

// or this might be a faster implementation, not that it matters so much
const mapValues = (obj, fn) =>
  Object.fromEntries(Object.entries(obj).map((key, val) => fn(val, key)))
```

and then this bit could just be:

```ts
setSampleSizes(mapValues(sizes, (size, contestId) =>
  sizes[contestId] === 'custom' ? customSampleSizes[contestId] : sizes[contestId]
))
```

Reply:
@jonahkagan I find the ES6 format a bit more self-documenting and readable, but that's probably just me. And this will likely change a bit when you redo the api. I'll look at it again when I refactor for that.